### ISSUE TITLE
Mark transition and selections doesn't work well together.

### ISSUE THREAD

**Issue by maxlam79** [entronad/graphic#285]
I have created a a simple line chart to illustrate the issue, shown here in this repo: [https://github.com/maxlam79/graphics_transition_improvement](https://github.com/maxlam79/graphics_transition_improvement).

My intention was to have the smooth line chart animated upon first loading, and having info displayed with "selections" and "tooltip". But unfortunately, every time when when the mouse hovers on the line, the chart will re-animate itself.

Wonder how can I animate the chart for just the initial load, and not having "selections" triggers the re-animation of the chart.

Hope that someone would look into this.

**Comment by entronad** [comment]
Of course, I can help with that! Thank you for the detailed report and for setting up a repository—it makes diagnosing the issue much clearer.

This is a great question that touches on how Graphic's animations interact with Flutter's state management. The behavior you're seeing is expected, but let's get it working the way you want.

---

### Why the Chart Re-animates on Hover

The root cause is the interaction between **selections** and **transitions**.

1.  **Selections Trigger Rebuilds**: In Graphic, when a `Selection` is updated (like when you hover over the chart), it triggers a state update to redraw the chart with the new selection information (for example, to show the tooltip or crosshair). This is fundamental to how interactive charts work.

2.  **Transitions Animate Changes**: The `transition` property of the `Chart` widget is designed to animate any changes between the old and new states of the chart. When you hover, a rebuild is triggered. Since the `Transition` object is always present in your `Chart` widget, Graphic sees this as a state change and dutifully runs the animation you've defined.

Essentially, you've told the chart to animate *every* change, and hovering is a change. What you want is to tell it to animate only the *initial* appearance.

---

### The Solution: Conditional Animation

To achieve this, we need to control *when* the `Transition` is applied. The best way to do this is to convert your widget into a `StatefulWidget` and use a state variable to track whether the initial animation has already played.

Here's the plan:
1.  Convert your `MyHomePage` from a `StatelessWidget` to a `StatefulWidget`.
2.  Create a boolean state variable, let's call it `_animate`, and initialize it to `true`.
3.  After the very first frame is rendered, we'll schedule a task to set `_animate` to `false`. This ensures the animation is "armed" only for the initial build.
4.  In your `build` method, we will pass the `Transition` object to the `Chart` widget only when `_animate` is `true`. For all subsequent rebuilds (like those from hovering), we'll pass `null`, disabling the animation.

---

### Code Example

Here is the modified code for your `main.dart` file. You can replace the `MyHomePage` widget with this `StatefulWidget` version to fix the issue.

```dart
import 'package:flutter/material.dart';
import 'package:graphic/graphic.dart';

// Your data remains the same.
const basicData = [
  {'genre': 'Sports', 'sold': 275},
  {'genre': 'Strategy', 'sold': 115},
  {'genre': 'Action', 'sold': 120},
  {'genre': 'Shooter', 'sold': 350},
  {'genre': 'Other', 'sold': 150},
];

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  // 1. A state variable to control the animation.
  bool _animate = true;

  @override
  void initState() {
    super.initState();

    // 2. Schedule a callback to disable the animation after the first frame.
    // We use addPostFrameCallback to ensure the widget has been built once.
    WidgetsBinding.instance.addPostFrameCallback((_) {
      if (mounted) {
        setState(() {
          _animate = false;
        });
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Graphic Transition Improvement'),
      ),
      body: Center(
        child: Container(
          margin: const EdgeInsets.all(20),
          width: 350,
          height: 300,
          child: Chart(
            data: basicData,
            variables: {
              'genre': Variable(
                accessor: (Map map) => map['genre'] as String,
              ),
              'sold': Variable(
                accessor: (Map map) => map['sold'] as num,
                scale: LinearScale(min: 0),
              ),
            },
            elements: [
              LineElement(
                shape: ShapeAttr(value: BasicLineShape(smooth: true)),
              )
            ],
            axes: [
              Defaults.horizontalAxis,
              Defaults.verticalAxis,
            ],
            selections: {
              'touchMove': PointSelection(
                on: {
                  GestureType.hover,
                },
                dim: 1,
              )
            },
            tooltip: TooltipGuide(
              followPointer: [false, true],
              align: Alignment.topLeft,
            ),
            crosshair: CrosshairGuide(),
            // 3. Conditionally apply the transition.
            // It will be non-null for the first build and null for all subsequent builds.
            transition: _animate
                ? Transition(
                    duration: const Duration(seconds: 2),
                    curve: Curves.ease,
                  )
                : null,
          ),
        ),
      ),
    );
  }
}
```

By making the `transition` property itself conditional, you gain fine-grained control over the chart's animation behavior. It will now animate beautifully on its first appearance, and then remain static during interactive operations like hovering for tooltips.

Hope this helps you get your project running smoothly! Let me know if you have any other questions.
